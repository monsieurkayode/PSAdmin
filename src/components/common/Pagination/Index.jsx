import React from 'react';
import ReactPaginate from 'react-paginate';
import { number } from 'prop-types';

const Pagination = ({ pageCount, page, handlePageChange }) => ( // eslint-disable-line
  <div style={{ textAlign: 'center' }}>
    <ReactPaginate
      breakLabel={<span>...</span>}
      breakClassName="break-me"
      pageCount={pageCount}
      initialPage={page - 1}
      marginPagesDisplayed={2}
      pageRangeDisplayed={3}
      onPageChange={handlePageChange}
      disableInitialCallback
      containerClassName="pagination"
      subContainerClassName="pages pagination"
      activeClassName="active"
    />
  </div>
);

Pagination.propTypes = {
  pageCount: number.isRequired,
  page: number.isRequired
};

export default Pagination;
