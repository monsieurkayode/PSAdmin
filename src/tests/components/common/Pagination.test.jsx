import React from 'react';
import { mount } from 'enzyme';
import Pagination from '../../../components/common/Pagination/Index';

const setup = () => {
  const props = {
    pageCount: 1,
    page: 2,
    handlePageChange: jest.fn()
  };

  return mount(<Pagination {...props} />);
};

describe('Pagination Component', () => {
  const wrapper = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
