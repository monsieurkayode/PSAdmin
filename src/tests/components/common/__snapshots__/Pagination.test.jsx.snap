// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component renders correctly 1`] = `
<Pagination
  handlePageChange={[MockFunction]}
  page={2}
  pageCount={1}
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <PaginationBoxView
      activeClassName="active"
      breakClassName="break-me"
      breakLabel={
        <span>
          ...
        </span>
      }
      containerClassName="pagination"
      disableInitialCallback={true}
      disabledClassName="disabled"
      initialPage={1}
      marginPagesDisplayed={2}
      nextClassName="next"
      nextLabel="Next"
      onPageChange={[MockFunction]}
      pageCount={1}
      pageRangeDisplayed={3}
      previousClassName="previous"
      previousLabel="Previous"
      subContainerClassName="pages pagination"
    >
      <ul
        className="pagination"
      >
        <li
          className="previous"
        >
          <a
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            Previous
          </a>
        </li>
        <PageView
          activeClassName="active"
          key="0"
          onClick={[Function]}
          page={1}
          selected={false}
        >
          <li>
            <a
              aria-current={null}
              aria-label="Page 1"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            >
              1
            </a>
          </li>
        </PageView>
        <li
          className="next"
        >
          <a
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            Next
          </a>
        </li>
      </ul>
    </PaginationBoxView>
  </div>
</Pagination>
`;
